leastsquare.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+
+
+def calculate_weight(basis_function, output_labels):
+
+    # For MW = S
+    S = np.dot(basis_function, output_labels)
+    M = np.dot(basis_function, np.transpose(basis_function))
```

Comment:
Another way to do this is as below, not sure which is more pythonic:
```python
M = basis_function@basis_function.T
```

Reply:
Simplified implementation. Implemented